src/controller/CHIPDeviceController.h

```diff
@@ -144,6 +144,16 @@ class DLL_EXPORT DeviceController : public SessionRecoveryDelegate, public Abstr
      */
     virtual CHIP_ERROR Shutdown();
 
+    SessionManager * SessionMgr()
```

Comment:
I think we should strongly consider removing this and changing the one consumer we have (chip-tool) to get the system state from the device controller factory and get the session manager from there...

Failing that, a name that more clearly indicates that null can be returned (e.g. GetSessionMgr), or a signature that can fail, would be good...